src/inlineSuggestions/highlightStackAttributions.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  Range,
+  Selection,
+  WorkspaceConfiguration,
+  window,
+  workspace,
+  env,
+  Uri
+} from "vscode";
+
+import fetch from "node-fetch";
+
+export default async function highlightStackAttributions(): Promise<void> {
+  const document = window.activeTextEditor?.document
+  if (!document) return;
+
+  type Config = WorkspaceConfiguration & {
+    attributionWindowSize: number;
+  };
+  const config: Config = workspace.getConfiguration("HuggingFaceCode") as Config;
+  const { attributionWindowSize } = config;
+
+  // get cursor postion and offset
+  const cursorPosition = window.activeTextEditor?.selection.active;
+  if (!cursorPosition) return;
+  const cursorOffset = document.offsetAt(cursorPosition);
+
+  const start = Math.max(0, cursorOffset - attributionWindowSize);
+  const end = Math.min(document.getText().length, cursorOffset + attributionWindowSize);
+
+  // Select the start to end span
+  if (!window.activeTextEditor) return;
+  window.activeTextEditor.selection = new Selection(document.positionAt(start), document.positionAt(end));
+  // new Range(document.positionAt(start), document.positionAt(end));
+  
+
+  const text = document.getText();
+  const textAroundCursor = text.slice(start, end);
+
+  const url = "https://stack.dataportraits.org/overlap";
+  const body = { document: textAroundCursor };
+
+  // notify user request has started
+  void window.showInformationMessage("Searching for nearby code in the stack...");
+    
+  const resp = await fetch(url, {
+    method: "POST",
+    body: JSON.stringify(body),
+    headers: { "Content-Type": "application/json" },
+  });
+
+if(!resp.ok){
+    return;
+}
+
```

Comment:
```suggestion
    if(!resp.ok){
        return;
    }
```suggestion
if(!resp.ok){
    return;
}

```

```